src/eventgrid/azext_eventgrid/_help.py

```diff
@@ -137,7 +188,566 @@
 examples:
   - name: Update the properties of an existing domain.
-    text: az eventgrid domain update -g rg1 --name domain1 --allow-traffic-from-all-ips false --inbound-ip-rules 10.0.0.0/8 Allow --inbound-ip-rules 10.2.0.0/8 Allow --tags Dept=IT
+    text: az eventgrid domain update -g rg1 --name domain1 --sku Basic --identity noidentity --public-network-access Disabled --inbound-ip-rules 10.0.0.0/8 Allow --inbound-ip-rules 10.2.0.0/8 Allow --tags Dept=IT
+"""
+
+helps['eventgrid partner'] = """
+type: group
+short-summary: Manage partner resources.
+"""
+
+helps['eventgrid partner registration'] = """
+type: group
+short-summary: Manage partner registrations.
+"""
+
+helps['eventgrid partner registration create'] = """
+type: command
+short-summary: Create a new partner registration.
+examples:
+  - name: Create a new partner registration with basic information.
+    text: az eventgrid partner registration create -g rg1 --name partnerRegistrationName1 --partner-name Contoso --resource-type-name Accounts --authorized-subscription-ids 533ad9de-25db-46e2-b94a-d00c37cf022b 05aa2228-7d34-4635-922d-2b582c422445
+  - name: Create a new partner registration with partner information.
+    text: az eventgrid partner registration create -g rg1 --name partnerRegistrationName1 --partner-name Contoso --resource-type-name Accounts --authorized-subscription-ids 533ad9de-25db-46e2-b94a-d00c37cf022b 05aa2228-7d34-4635-922d-2b582c422445 --description ExampleDescription --display-name ExampleDisplayName1 --logo-uri https://www.example.com/logo.png --setup-uri https://www.example.com
+"""
+
+helps['eventgrid partner registration list'] = """
+type: command
+short-summary: List all partner registrations in specific resource group or all under the specified azure subscription.
+examples:
+  - name: List all partner registrations in the current Azure subscription.
+    text: az eventgrid partner registration list
+  - name: List all partner registrations in a resource group.
+    text: az eventgrid partner registration list -g rg1
+  - name: List all partner registrations in a resource group whose name contains the pattern "XYZ"
+    text: az eventgrid partner registration list -g rg1 --odata-query "Contains(name, 'XYZ')"
+  - name: List all partner registrations in a resource group except the partner registration with name "name1"
+    text: az eventgrid partner registration list -g rg1 --odata-query "NOT (name eq 'name1')"
+"""
+
+helps['eventgrid partner namespace'] = """
+type: group
+short-summary: Manage partner namespaces.
+"""
+
+helps['eventgrid partner namespace create'] = """
+type: command
+short-summary: Create a partner namespace.
+examples:
+  - name: Create a new partner namespace.
+    text: az eventgrid partner namespace create -g rg1 --name namespaceName1 -l westus2 --partner-registration-id 795c9f2f-6d2d-42ff-a570-42fd3043192c
```

Comment:
wjust curious , why not support name also? since when creating partner registration, user input name there. it's convinient to support registration name in namespace

Reply:
These are separate tracked resources. Partner registrations is global and partner needs to create it once, while namespace is regional and partner needs to create it in any location he is interested in (not necessarily all).